### PR TITLE
Bugfix/period exception

### LINKED ISSUE #228: Handler PeriodException when getting bucket metrics with no data provided by InputStream
the change [remove "flush timer"](https://github.com/ns1labs/pktvisor/pull/192) fixed issues on bucket because it creates a new timer thread to generate empty buckets.

However, for a policy with multiple handlers, If one handler does not receive inputs and the other do, when we try to get bucket it will generate a exception for this particular handler and REST API will return error (425).

### PR DESCRIPTION
**Pull request by leoparente**

### CODE REVIEW

File: src/AbstractMetricsManager.h [hunk 1]
```diff
@@ -440,6 +440,16 @@ class AbstractMetricsManager
         }
     }
 
+    void shift_bucket(timespec stamp)
```
Comment (weyrick) [inline src/AbstractMetricsManager.h]: please refactor name to `check_period_shift`

File: src/InputStream.h [hunk 2]
```diff
@@ -15,6 +15,9 @@ class InputStream : public AbstractRunnableModule
     mutable std::shared_mutex _input_mutex;
     std::vector<const Policy *> _policies;
 
+protected:
+    static constexpr uint8_t RUNNING_NOTIFY_INTERVAL = 30; // in seconds
```
Comment (weyrick) [inline src/InputStream.h]: please refactor names from "running" to "heartbeat", so `HEARTBEAT_INTERVAL`, `heartbeat_signal`, etc.

File: src/inputs/dnstap/DnstapInputStream.cpp [hunk 3]
```diff
@@ -224,6 +240,7 @@ void DnstapInputStream::_create_frame_stream_tcp_socket()
 
     // spawn the loop
     _io_thread = std::make_unique<std::thread>([this] {
+        _timer->start(uvw::TimerHandle::Time{1000}, uvw::TimerHandle::Time{RUNNING_NOTIFY_INTERVAL * 1000});
```
Comment (weyrick) [inline src/inputs/dnstap/DnstapInputStream.cpp]: is there a spot where we need to `stop()` or `close()` or something, when the input is stopped (not just on error event?)

**Review by weyrick — CHANGES_REQUESTED** [review]

File: src/AbstractMetricsManager.h (hunk 1 above)
Comment (leoparente) [reply src/AbstractMetricsManager.h]: done

File: src/InputStream.h (hunk 2 above)
Comment (leoparente) [reply src/InputStream.h]: done

File: src/inputs/dnstap/DnstapInputStream.cpp (hunk 3 above)
Comment (leoparente) [reply src/inputs/dnstap/DnstapInputStream.cpp]: based on their unit tests it seems we dont actually need to stop and close it. but I did it on async event as well

Comment (leoparente) [reply src/inputs/dnstap/DnstapInputStream.cpp]: done

**Review by weyrick — APPROVED** [review]

**Pull request merged**